modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/web/internal/context/test/SegmentsDisplayContextTest.java

```diff
@@ -85,13 +119,161 @@ public void setUp() throws Exception {
 			null);
 
 		_serviceTracker.open();
+
+		_user = UserTestUtil.addUser();
+
+		_mockLiferayPortletRenderRequest = _renderPortlet();
 	}
 
 	@After
 	public void tearDown() {
 		_serviceTracker.close();
 	}
 
+	@Test
+	public void testGetActionDropdownItems() {
+		List<DropdownItem> dropdownItems = ReflectionTestUtil.invoke(
+			_mockLiferayPortletRenderRequest.getAttribute(
+				"SEGMENTS_DISPLAY_CONTEXT"),
+			"getActionDropdownItems", new Class<?>[0]);
+
+		Assert.assertEquals(dropdownItems.toString(), 1, dropdownItems.size());
+	}
+
+	@Test
+	public void testGetAssignUserRolesDataMap() throws Exception {
+		SegmentsEntry segmentsEntry = _getSegmentEntry();
+
+		HashMap<String, Object> assignUserRolesDataMap = _invokeMethod(
+			"getAssignUserRolesDataMap", segmentsEntry);
+
+		Assert.assertEquals(
+			segmentsEntry.getSegmentsEntryId(),
+			assignUserRolesDataMap.get("segmentsEntryId"));
+
+		String itemSelectorURL = String.valueOf(
+			assignUserRolesDataMap.get("itemSelectorURL"));
+
+		itemSelectorURL = HttpComponentsUtil.decodeURL(itemSelectorURL);
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray(
```

Comment:
We convert it to string, just to compare the sctring as jsonArray.toString(), maybe it is better to use _getExcludedRoleNames() directly in the assert method?